src/blocks/calendar.rs

```diff
@@ -0,0 +1,611 @@
+                let credentials = if let Some(path) = &oauth2.credentials_path {
+                    util::deserialize_toml_file(path.expand()?.to_string())
+                        .error("Failed to read oauth2 credentials file")?
+                } else {
+                    oauth2.credentials.clone()
+                };
+                let OAuth2Credentials {
+                    client_id: Some(client_id),
+                    client_secret: Some(client_secret),
+                } = credentials
+                else {
+                    return Err(Error::new("Oauth2 credentials are not configured"));
+                };
+                let auth_url =
+                    AuthUrl::new(oauth2.auth_url.clone()).error("Invalid authorization url")?;
+                let token_url =
+                    TokenUrl::new(oauth2.token_url.clone()).error("Invalid token url")?;
+
+                let flow = OAuth2Flow::new(
+                    ClientId::new(client_id),
+                    ClientSecret::new(client_secret),
+                    auth_url,
+                    token_url,
+                    oauth2.redirect_port,
+                );
+                let token_store =
+                    TokenStore::new(Path::new(&oauth2.auth_token.expand()?.to_string()));
+                auth::Auth::oauth2(flow, token_store, oauth2.scopes.clone())
+            }
+        };
+        Ok(Self {
+            client: Client::new(
+                Url::parse(&config.url).error("Invalid CalDav server url")?,
+                auth,
+            ),
+            config,
+        })
+    }
+
+    async fn get_next_events(
+        &mut self,
+        within: Duration,
+    ) -> Result<OverlappingEvents, CalendarError> {
+        let calendars: Vec<_> = self
+            .client
+            .calendars()
+            .await?
+            .into_iter()
+            .filter(|c| self.config.calendars.is_empty() || self.config.calendars.contains(&c.name))
+            .collect();
+        let mut events: Vec<Event> = vec![];
+        for calendar in calendars {
+            let calendar_events: Vec<_> = self
+                .client
+                .events(
+                    &calendar,
+                    Utc::now()
+                        .date_naive()
+                        .and_hms_opt(0, 0, 0)
+                        .expect("A valid time")
+                        .and_local_timezone(Local)
+                        .earliest()
+                        .expect("A valid datetime")
+                        .to_utc(),
+                    Utc::now() + within,
+                )
+                .await?
+                .into_iter()
+                .filter(|e| {
+                    let not_started = e.start_at.is_some_and(|d| d > Utc::now());
+                    let is_ongoing = e.start_at.is_some_and(|d| d < Utc::now())
+                        && e.end_at.is_some_and(|d| d > Utc::now());
+                    not_started || is_ongoing

```

Comment:
Shouldn't all of the events received be not_started or ongoing?

Reply:
The block retrieves events from the start of the day to make the ongoing visible after `Utc::now`.  It is not a perfect solution, because you can have events that last multiple days and started days before. 
I usually use this block to be warned about my next meeting of the day and I want to know if there is an ongoing meeting for which I missed the warning. 
If it is an event that lasts days, I should already know about it so I do not care if I do not see it as an ongoing event in the block.